editor/src/components/common/shared-strategies/convert-to-flex-strategy.ts

```diff
@@ -170,18 +188,61 @@ export function convertLayoutToFlexCommands(
   })
 }
 
-function ifElementIsFragmentFirstConvertItToFrame(
+function ifElementIsFragmentLikeFirstConvertItToFrame(
   metadata: ElementInstanceMetadataMap,
-  elementPathTree: ElementPathTrees,
+  allElementProps: AllElementProps,
+  elementPathTrees: ElementPathTrees,
   target: ElementPath,
 ): Array<CanvasCommand> {
-  return convertFragmentToFrame(
+  const type = getElementFragmentLikeType(
     metadata,
-    elementPathTree,
-    {},
+    allElementProps,
+    elementPathTrees,
     target,
-    'convert-even-if-it-has-static-children',
+    'sizeless-div-considered-fragment-like',
   )
+  const elementInstace = MetadataUtils.findElementByElementPath(metadata, target)
+  if (
+    elementInstace == null ||
+    isLeft(elementInstace.element) ||
+    !isJSXElementLike(elementInstace.element.value)
+  ) {
+    return []
+  }
+
+  if (type == 'fragment' || type == 'sizeless-div') {
```

Comment:
is this narrowing going to be a problem? should this become a helper in `fragment-like-helpers.ts`?

Reply:
I think the narrowing could work as long as this function is local to the convert to flex strategy (since here we only want to return actual commands if the element to be converted is a fragment or a sizeless div), but if this function would become a general helper, this would have to be thought through.

Would it work to change the name of the function to ifElementIsFragmentOrSizelessDivFirstConvertItToFrame?